src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -632,6 +632,91 @@ private void CleanUp()
         #endregion IDisposable Members
     }
 
+    /// <summary>
+    /// Implements ConvertTo-CliXml command.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2243:AttributeStringLiteralsShouldParseCorrectly")]
+    [Cmdlet(VerbsData.ConvertTo, "CliXml", HelpUri = "")]
+    [OutputType(typeof(string))]
+    public sealed class ConvertToClixmlCommand : PSCmdlet
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets input objects to be converted to CliXml object.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets depth of serialization.
+        /// </summary>
+        [Parameter]
+        [ValidateRange(1, int.MaxValue)]
+        public int Depth { get; set; } = 1;
```

Comment:
This was interesting. I wonder how it worked in `Export-CliXml` since `ValidateRange` should error on default of 0?

Reply:
I haven't looked at the code, but I found in my notes the following snippet, implying that `Export-CliXml` uses `2` as the depth.

```powershell
class Bar {
  [int] $Baz = 42
}
class Foo {
  [Bar] $Baz = [Bar]::new()
}

$o = [Foo]::new()

[System.Management.Automation.PSSerializer]::Serialize($o)

Write-Host '---'

$o | Export-Clixml t.xml

Get-Content t.xml
```

Reply:
@ArmaanMcleod Could you please re-check? I believe it is more safe use explicit default here and in other cmdlets.

Reply:
So Export-CliXml doesn't set an explicit default for the parameter,

```pwsh
# Export-CliXml
[Parameter]
[ValidateRange(1, int.MaxValue)]
public int Depth { get; set; }
```

I wonder if it's doing this in the background somewhere, couldn't see in the code where its being done, or if internal API serializer has default depth of 2 somewhere.

But I guess in this case if it observed to have default depth of 2 then I can do the same for `ConvertTo-CliXml`. Might just be easier to be explicit and set a default to 2 to the parameter directly.

There is some interesting code for `Export-CliXml` which checks Depth == 0, which is for when depth is not specified since 0 is default for `int`, but I think in the background the depth is really 2 from default. 

Docs seem to also indicate default depth is 2: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/export-clixml?view=powershell-7.4#-depth

Reply:
> couldn't see in the code where its being done, or if internal API serializer has default depth of 2 somewhere.

Is the default in .Net?
-----

Have we tests for the default for both cmdlets? If no, please add to avoid regressions in future.

Reply:
Could not find in .Net API but I've added tests for default of 2. Should be ok for now to avoid regression.